src/main/resources/templates/client/cypher/detail.html

```diff
@@ -0,0 +1,162 @@
+                    </a>
+                </div>
+            </th:block>
+        </th:block>
+
+        <div class="pb-2">
+            <b>Adresa</b>
+            <br/>
+            <div th:text="${cypher.location}"></div>
+            <br/>
+            <div th:text="${cypher.placeDescription}"></div>
+            <br/>
+            <div>ZDE MAPA</div>
+            <button href="#" class="btn-white" type="submit">Zobrazit na mapě</button>
+        </div>
+        <br/>
+
+            <th:block th:if="${status == 'PENDING'}">
+                <b>Nápověda</b>
+
+                <th:block th:if="${not #lists.isEmpty(hintsTaken)}">
+                    <div th:each="hintTaken : ${hintsTaken}">
+                        <p th:text="|Nápověda (-${hintTaken.hint.value}b.): ${hintTaken.hint.text}|"></p>
+                    </div>
+                </th:block>
+                <form action="/hint" th:object="${cypher}">
+                    <input type="hidden"  th:field="*{cypherId}" >
+                    <input type="hidden"  th:field="*{hints}" >
+                    <input type="hidden"  th:field="*{name}" >
+                    <input type="hidden"  th:field="*{stage}" >
+                    <input type="hidden"  th:field="*{placeDescription}" >
+                    <input type="hidden"  th:field="*{location}" >
+                    <input type="hidden"  th:field="*{codeword}" >
+                    <input type="hidden"  th:field="*{bonusContent}" >
+                    <button type="submit" class="btn-white">Získat nápovědu</button>
+                </form>
+                <br/>
+            </th:block>
+
+        <th:block th:if="${status == 'PENDING'}">
+            <form onsubmit="return confirm('Opravdu to chcete vzdát?')" action="/cypher/giveUp" th:object="${cypher}" method="post">
+                <input type="hidden"  th:field="*{cypherId}" >
+                <input type="hidden"  th:field="*{hints}" >
+                <input type="hidden"  th:field="*{name}" >
+                <input type="hidden"  th:field="*{stage}" >
+                <input type="hidden"  th:field="*{placeDescription}" >
+                <input type="hidden"  th:field="*{location}" >
+                <input type="hidden"  th:field="*{codeword}" >
+                <input type="hidden"  th:field="*{bonusContent}" >
+                <br/>
+                <button class="btn-red" type="submit" onclick="">Vzdát šifru</button>
+            </form>
+        </th:block>
+
+        <script>
+            function showBonus() {
+                var bonusInfo = document.getElementById("bonusInfo");
+                bonusInfo.style.display = "block";
+                var bonusButton = document.getElementById("bonusButton");
+                bonusButton.style.display = "none";
+            }
+
+            function checkCodeword(form) {
+                if(form.codeword.value == form.guess.value) {
```

Comment:
V odpovedi nesmi prijit spravna odpoved! Server musi vratit pouze true nebo false. Nejsem si presne jisty jak to tady funguje, ale tato kontrola musi byt jen na serveru a tady urcite ne. Predelat tak, ze to tady nebude. Pokud zada spravnou odpoved -> Prekresli se stranka a bude zobrazena varianta pro solved cypher a pokud ne, zobrazi se, ze to nebylo spravne reseni.